src/main/kotlin/uk/gov/justice/digital/hmpps/managerecallsapi/service/RecallAuditService.kt

```diff
@@ -71,51 +75,51 @@ class RecallAuditService(
         recallField.getAnnotation(Column::class.java).name
       } else {
         if (
-          persister.getSubclassPropertyColumnNames(fieldPath) != null &&
-          persister.getSubclassPropertyColumnNames(fieldPath).isNotEmpty()
+          persister.getSubclassPropertyColumnNames(fieldPath.value) != null &&
+          persister.getSubclassPropertyColumnNames(fieldPath.value).isNotEmpty()
         ) {
-          persister.getSubclassPropertyColumnNames(fieldPath)
+          persister.getSubclassPropertyColumnNames(fieldPath.value)
         } else {
-          persister.getPropertyColumnNames(fieldName.getNameFromPath())
+          persister.getPropertyColumnNames(fieldPath.getFieldName().value)
         }[0]
       }
     )
   }
 
-  private fun getRecallFieldAndPath(fieldName: FieldName): Pair<Field, String> {
-    val fieldPathParts = fieldName.value.split(".")
+  private fun getRecallField(fieldPath: FieldPath): Field {
+    val fieldPathParts = fieldPath.value.split(".")
     var field = Recall::class.java.getDeclaredField(fieldPathParts[0])
     for (i in 1 until fieldPathParts.size) {
       field = field.type.getDeclaredField(fieldPathParts[i])
     }
-    return Pair(field, fieldName.value)
+    return field
   }
 
-  private fun getFieldNameForColumn(columnName: ColumnName): FieldName =
+  private fun getFieldPathForColumn(columnName: ColumnName): FieldPath =
     Recall::class.java.declaredFields.flatMap { field ->
-      val propCols = persister.getPropertyColumnNames(field.name).map { ColumnName(it) to FieldName(field.name) }
+      val propCols = persister.getPropertyColumnNames(field.name).map { ColumnName(it) to FieldPath(field.name) }
       if (propCols.size == 1) {
         propCols
       } else {
         field.type.declaredFields.flatMap { childField ->
           val childFieldPath = "${field.name}.${childField.name}"
           val childProbCols = persister.getPropertyColumnNames(childFieldPath)
```

Comment:
`childPropCols` not `childProbCols`?

Reply:
child PropertyColumns (from getPropertyColumnNames)